beetsplug/musicbrainz.py

```diff
@@ -469,19 +469,30 @@ def track_info(
                 "artist-relation-list", ()
             ):
                 if "type" in artist_relation:
+                    alias = _preferred_alias(
+                        artist_relation["artist"].get("alias-list", ())
+                    )
+
+                    # Get artist name alias
+                    if alias:
+                        cur_artist_rel_name = alias["alias"]
+                    else:
+                        cur_artist_rel_name = artist_relation["artist"]["name"]
+
+                    artist = artist_relation["artist"]
+                    alias = _preferred_alias(artist.get("alias-list", ()))
+
+                    if alias:
+                        cur_artist_rel_sort_name = alias["sort-name"]
+                    else:
+                        cur_artist_rel_sort_name = artist["sort-name"]
+
                     type = artist_relation["type"]
                     if type == "lyricist":
-                        lyricist.append(artist_relation["artist"]["name"])
+                        lyricist.append(cur_artist_rel_name)
```

Comment:
I think this logic an be simplified to something like the following

```suggestion
                    artist = artist_relation["artist"]
                    alias = _preferred_alias(artist.get("alias-list", ()))

                    artist = alias["alias"] if alias else artist["name"]
                    artist_sort = (alias if alias else artist)["sort-name"]

                    type = artist_relation["type"]
                    if type == "lyricist":
                        lyricist.append(artist)
                    elif type == "composer":
                        composer.append(artist)
                        composer_sort.append(artist_sort)

```